engine/gamesys/src/gamesys/scripts/script_box2d_body.cpp

```diff
@@ -0,0 +1,885 @@
+        return 1;
+    }
+
+    static int Body_GetLocalPoint(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+        b2Body* body = CheckBody(L, 1);
+        b2Vec2 p = CheckVec2(L, 1, GetPhysicsScale());
+        dmScript::PushVector3(L, FromB2(body->GetLocalPoint(p), GetInvPhysicsScale()));
+        return 1;
+    }
+
+    static int Body_GetLocalVector(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+        b2Body* body = CheckBody(L, 1);
+        b2Vec2 p = CheckVec2(L, 1, GetPhysicsScale());
+        dmScript::PushVector3(L, FromB2(body->GetLocalVector(p), GetInvPhysicsScale()));
+        return 1;
+    }
+
+    static int Body_GetLinearVelocityFromWorldPoint(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+        b2Body* body = CheckBody(L, 1);
+        b2Vec2 p = CheckVec2(L, 1, GetPhysicsScale());
+        dmScript::PushVector3(L, FromB2(body->GetLinearVelocityFromWorldPoint(p), GetInvPhysicsScale()));
+        return 1;
+    }
+
+    static int Body_GetLinearVelocityFromLocalPoint(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+        b2Body* body = CheckBody(L, 1);
+        b2Vec2 p = CheckVec2(L, 1, GetPhysicsScale());
+        dmScript::PushVector3(L, FromB2(body->GetLinearVelocityFromLocalPoint(p), GetInvPhysicsScale()));
+        return 1;
+    }
+
+    static int Body_GetWorld(lua_State *L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+        b2Body* body = CheckBody(L, 1);
+        PushWorld(L, body->GetWorld());
+        return 1;
+    }
+
+    static int Body_GetNext(lua_State *L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+        b2Body* body = CheckBody(L, 1);
+        b2Body* next = body->GetNext();
+        if (next)
+            PushBody(L, next);
+        else
+            lua_pushnil(L);
+        return 1;
+    }
+
+    static int Body_Dump(lua_State *L)
+    {
+        DM_LUA_STACK_CHECK(L, 0);
+        b2Body* body = CheckBody(L, 1);
+        body->Dump();
+        return 0;
+    }
+
+    static int Body_tostring(lua_State *L)
+    {
+        b2Body* body = CheckBody(L, 1);
+        lua_pushfstring(L, "Box2D.%s = %p", BOX2D_TYPE_NAME_BODY, body);
+        return 1;
+    }
+
+    static int Body_eq(lua_State *L)
+    {
+        b2Body* a = ToBody(L, 1);
+        b2Body* b = ToBody(L, 2);
+        lua_pushboolean(L, a && b && a == b);
+        return 1;
+    }
+
+    static int Body_newindex(lua_State *L)
+    {
+        return luaL_error(L, BOX2D_TYPE_NAME_BODY " does not support adding new elemnents");
+    }
+
+    static const luaL_reg Body_methods[] =
+    {
+        {0,0}
+    };
+
+    static const luaL_reg Body_meta[] =
+    {
+        {"__tostring", Body_tostring},
+        {"__eq", Body_eq},
+
+        {"get_position", Body_GetPosition},
+        {"set_transform", Body_SetTransform},
+
+        //{"get_user_data", Body_GetUserData}, - could return the game object id ?
+        //{"set_user_data", Body_SetUserData}, - could attach the body to a game object?
```

Comment:
Getting the id of the game object is useful, and it should be possible to set the id as well, with some extra component functions to make sure that the game+physics objects life times are considered.